primary/src/core.rs

```diff
@@ -423,6 +423,16 @@ impl Core {
                 .send((parents, certificate.round(), certificate.epoch()))
                 .await
                 .map_err(|_| DagError::ShuttingDown)?;
+
+            let before = self.cancel_handlers.len();
+            if certificate.round() > 0 {
+                self.cancel_handlers
```

Comment:
Nit: if we want to track the debate on whether this bound of `certificate.round()` can be fine-tuned `certificate.round()-1`, a comment here or an issue would do nicely.